src/Registry.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+error AlreadyAdded();
+
+/// @title  Registry contract :: Adding popular token's data :: DAO Owned 
+/// @notice This contract can be called by our main contract for fetching tokenData
+
+contract Registry {
+
+    ///  Address of the DAO
+    address immutable owner_DAO;
+
+    /// This struct contains necessary info about an ERC20 token
+    /// What more params could be added?
+    struct Token{
+        string name;
+        string symbol;
+        uint8 UID;
+        uint8 decimals;
+    }
+    
+    /// Registry mapping ERC20 token address to its metadata
+    mapping(address=>Token) tokenData;
```

Comment:
We can fetch this info from chain directly from ERC20 contract, I don't think we need to store it here

Reply:
I thought we would make the NFT design specific to each popular ERC20 and this struct would contain additional data ( not basic ERC20 info ) of each erc20 address.